Refactor Surface / SurfaceInterface and backend.

sletta opened the pull request.

sletta committed:
Remove unused file: openglcontext.h

sletta committed:
Refactor Surface / SurfaceInterface and backend.

This only fixes the SDL backend. Qt backend and SFHWC backends are left
out in the cold for now.

We're limiting ourselves to a single window in the implementation, but
the API doesn't enforce this so we're free to change it down the line.

However, the limiting makes our life very easy in the backends, as
we only have to deal with a single implementation class there, the
backend itself.

rburchell commented on include/backend/qt/qtbackend.h:
```diff
@@ -176,7 +176,7 @@ inline Surface *QtBackend::createSurface(SurfaceInterface *iface)
 {
     logd << __PRETTY_FUNCTION__ << surface << std::endl;
-    assert(surface);
+  onsi(surface);
```
onsi?

rburchell commented on include/util/standardsurface.h:
```diff
@@ -25,18 +25,16 @@
 public:
-    StandardSurfaceInterface()
+    StandardSurface()
     {
```
Should we enforce surfaces being a singleton here somehow?

rburchell reviewed (changes requested):
Seems like you are missing changes for SFHWC and Qt backends?

Looks mostly good. I question as to whether or not we should drop show/hide behaviour. Given that our primary target is a window that will be permanently visible, having these seems like another complication that we can potentially do without.

Creating a surface, then, would automatically show the window.

What do you think?

sletta committed:
Remove the Qt backend.

Maintaining multiple backends at this stage is a bit time consuming
and not really worth our effort. Plus, the SDL backend has similar
coverage as Qt anyway, so there is little to gain in having both
